neqo-transport/src/connection/mod.rs

```diff
@@ -1560,15 +1560,21 @@ impl Connection {
         let mut ack_eliciting = false;
         let mut probing = true;
         let mut d = Decoder::from(&packet[..]);
+        let mut decoded_frames = false;
         while d.remaining() > 0 {
             let f = Frame::decode(&mut d)?;
+            decoded_frames = true;
             ack_eliciting |= f.ack_eliciting();
             probing &= f.path_probing();
             let t = f.get_type();
             if let Err(e) = self.input_frame(path, packet.version(), packet.packet_type(), f, now) {
                 self.capture_error(Some(Rc::clone(path)), now, t, Err(e))?;
             }
         }
+        if !decoded_frames {
+            qerror!([self], "Received packet with no frames");
+            return Err(Error::ProtocolViolation);
+        }
```

Comment:
Why not check `packet.len()`?

This is going to pass if the packet contains only padding, which I think is OK, but you can make the code simpler.

```rust
packet.is_empty().then_some(()).ok_or(Error::ProtocolViolation)?;
```

I don't think that we need to use `qerror!()` here.  I'd stop at info level for errors that come from the other end.  Errors are best saved for stuff where we are badly stuck.  The code I suggest has no logging, which I think is also fine in this case.